File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.py
from collections import deque

N, M = map(int, input().split())
target = list(map(int, input().split()))
queue = deque([i for i in range(1, N+1)])

count = 0
for i in target:
    while True:
        if queue[0] == i:
            queue.popleft()
            break
        else:
            if queue.index(i) < len(queue)/2:
                while queue[0] != i:  # dq의 첫번째 인덱스가 i와 같아질때까지 반복
                    queue.append(queue.popleft())
                    count += 1
            else:  # 뽑아내려는 수의 위치 인덱스가 dq의 길이를 반으로 나눈것보다 클때는 오른쪽으로 움직여야 최소
                while queue[0] != i:
                    queue.appendleft(queue.pop())
                    count += 1
print(count)

File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.js
const input = require('fs').readFileSync(process.version === 'v18.17.1' ? 'input.txt' : '/dev/stdin')
  .toString().trim().split("\n");

const solution = () => {
  const [n, m] = input[0].split(" ").map(Number);
  const targetArr = input[1].split(" ").map(Number);

  let queue = Array.from({length: n}, (_, i) => i + 1);
  let count = 0;

  targetArr.forEach((target) => {
    let index = queue.indexOf(target);
    let left = index; // 왼쪽으로 이동하는 연산 횟수
    let right = queue.length - index; // 오른쪽으로 이동하는 연산 횟수
    count += Math.min(left, right); // 둘 중 최소값을 연산 횟수에 더함

    // 큐 회전 및 원소 제거
    if (left < right) {
      queue = [...queue.slice(index + 1), ...queue.slice(0, index)];
    } else {
      queue = [...queue.slice(index + 1), ...queue.slice(0, index)];
    }
  });

  console.log(count); // 수행한 연산 횟수 출력
};

solution();

